Cython/Utility/ObjectHandling.c

```diff
@@ -2929,3 +2929,13 @@ static PyObject *__Pyx_PyList_Pack(Py_ssize_t n, ...) {
     va_end(va);
     return l;
 }
+
+/////////////// LengthHint.proto //////////////////////////////////////
+
+#if !CYTHON_COMPILING_IN_LIMITED_API
+#define __Pyx_PyObject_LengthHint(o, defaultval) PyObject_LengthHint(o, defaultval)
+#else
+// We could reimplement it fully, however since it's only an optimization
+// the simpler version is to make it the default.
+#define __Pyx_PyObject_LengthHint(o, defaultval) defaultval
+#endif
```

Comment:
I understand that the Limited API case is the (potentially much) longer one, but it's still just a special case. If we add other special cases, I'd like to be able to just add them as second and third special case, not think about how they relate to the existing cases.

Also, I like leaving more space between the macro and its implementation, to visually separate the two.

```suggestion
#if CYTHON_COMPILING_IN_LIMITED_API
// We could reimplement it fully, however since it's only an optimization
// the simpler version is to make it the default.
#define __Pyx_PyObject_LengthHint(o, defaultval)  (defaultval)
#else
#define __Pyx_PyObject_LengthHint(o, defaultval)  PyObject_LengthHint(o, defaultval)
#endif
```